test_conformance/profiling/enqueueSyncFunctions.cpp

```diff
@@ -0,0 +1,287 @@
+    test_error(error, "Unable to create test buffer");
+
+    error = create_single_kernel_helper(context, &program, &kernel1, 1,
+                                        &test_kernel, "test1");
+    test_error(error, "Unable to create test kernel");
+
+    error = clSetKernelArg(kernel1, 0, sizeof(buffer1), &buffer1);
+    test_error(error, "Unable to set argument for test kernel");
+
+    error = create_single_kernel_helper(context, &program, &kernel2, 1,
+                                        &test_kernel, "test2");
+    test_error(error, "Unable to create test kernel");
+
+    error = clSetKernelArg(kernel2, 0, sizeof(buffer2), &buffer2);
+    test_error(error, "Unable to set argument for test kernel");
+
+    error = create_single_kernel_helper(context, &program, &kernel3, 1,
+                                        &test_kernel, "test1");
+    test_error(error, "Unable to create test kernel");
+
+    error = clSetKernelArg(kernel3, 0, sizeof(buffer3), &buffer3);
+    test_error(error, "Unable to set argument for test kernel");
+
+    clEventWrapper events_list_set1[3] = { NULL, NULL, NULL };
+    clEventWrapper events_list_set2[3] = { NULL, NULL, NULL };
+    // run 1 set of ndrange commands
+    error = clEnqueueNDRangeKernel(queue_with_props, kernel1, 1, NULL,
+                                   global_work_size, NULL, 0, NULL,
+                                   &events_list_set1[0]);
+    error |= clEnqueueNDRangeKernel(queue_with_props, kernel2, 1, NULL,
+                                    global_work_size, NULL, 0, NULL,
+                                    &events_list_set1[1]);
+    error |= clEnqueueNDRangeKernel(queue_with_props, kernel3, 1, NULL,
+                                    global_work_size, NULL, 0, NULL,
+                                    &events_list_set1[2]);
+    test_error(error, "Unable to enqueue kernels in set 1");
+
+    error =
+        fn(queue_with_props, 3, &events_list_set1[0], &eventEnqueueMarkerSet1);
+    test_error(error, "Unable to enqueue sync command");
+
+    // run 2 set of ndrange commands
+    error = clEnqueueNDRangeKernel(queue_with_props, kernel1, 1, NULL,
+                                   global_work_size, NULL, 0, NULL,
+                                   &events_list_set2[0]);
+    error |= clEnqueueNDRangeKernel(queue_with_props, kernel2, 1, NULL,
+                                    global_work_size, NULL, 0, NULL,
+                                    &events_list_set2[1]);
+    error |= clEnqueueNDRangeKernel(queue_with_props, kernel3, 1, NULL,
+                                    global_work_size, NULL, 0, NULL,
+                                    &events_list_set2[2]);
+    test_error(error, "Unable to enqueue kernels in set 2");
+
+    error =
+        fn(queue_with_props, 3, &events_list_set2[0], &eventEnqueueMarkerSet2);
+    test_error(error, "Unable to enqueue sync command");
+
+    error = clWaitForEvents(1, &eventEnqueueMarkerSet2);
```

Comment:
Should clWaitForEvents be switched to eventEnqueueMarkerSet1 since that's what's being checked in the following lines, the command queue is out of order, and set 1 doesn't wait for set 2 to complete? Or maybe clFinish would be more readable?